gpflow/utilities/utilities.py

```diff
@@ -246,22 +248,118 @@ def clear_bijector(bijector, _, state):
     return input_module
 
 
-M = TypeVar('M', bound=tf.Module)
+def _get_by_name_index(parent: object, attr_str: str, index_str: str) -> object:
+    attr = getattr(parent, attr_str)
+    if index_str != "":
+        try:
+            index = int(index_str)
+        except ValueError:
+            raise ValueError(f"Invalid index passed in the path'")
+        return attr[index]
+    return attr
```

Comment:
```suggestion
    else:
        return attr
```

Reply:
I don't mind changing it, but this `else` doesn't add clarity and it is more or less a style preference.

Reply:
it makes it nicely parallel with the `_set_by_name_index` below though